tests/test_validate.py

```diff
@@ -107,3 +108,74 @@ def check(prurl):
     assert len(check(', https://flit.readthedocs.io/')) == 1
     # Name longer than 32 chars
     assert len(check('Supercalifragilisticexpialidocious, https://flit.readthedocs.io/')) == 1
+
+
+def test_read_classifiers_cached(monkeypatch):
+    def mock_get_cache_dir():
+
+        class MockChache:
+            def open(self, encoding):
+                return self
+            def __truediv__(self, other):
+                return self
+            def __enter__(self):
+                return ["A", "B", "C"]
+            def __exit__(self, *args):
+                pass
+
+        return MockChache()
+
+    monkeypatch.setattr(fv, "get_cache_dir", mock_get_cache_dir)
```

Comment:
I don't much like this kind of complex mocking. It ties the test closely to how the function is implemented - e.g. it assumes the code does `with path.open() as f:` and iterates over f, rather than something like `path.read_text()`.

For tests that deal with file access, I prefer to make a temporary directory and mock things to point to that. Pytest has a handy `tmp_path` fixture which creates a temp dir and gives the test a pathlib object for it.

Reply:
For `test_read_classifiers_cached()` and `test_download_and_cache_classifiers()` I use the suggested pytest `tmp_path` fixtures.

For the `test_download_and_cache_classifiers_with_unacessible_dir()` test I couldn't come up with a good idea without a custom mock class, but I think I stripped it down to a bare minimum.